src/rgw/rgw_op.cc

```diff
@@ -1250,8 +1250,21 @@ int RGWGetObj::handle_user_manifest(const char *prefix)
     return 0;
   }
 
+  r = RGWRados::Object::Read::range_to_ofs(s->obj_size, ofs, end);
+  if (r < 0) {
+    return r;
+  }
+
+  r = iterate_user_manifest_parts(s->cct, store, ofs, end,
+        pbucket_info, obj_prefix, bucket_acl, *bucket_policy,
+        &total_len, nullptr, nullptr,
+        nullptr, nullptr);
```

Comment:
yes.  For negative offsets we should get the slo/dlo obj_size first.  Then fix the range.
The first `iterate_user_manifest_parts()` get object's objsize. Then fix the negative ranges.
The second `iterate_user_manifest_parts()` get total_len
The third `iterate_user_manifest_parts()` read data.